sycl/doc/design/SharedLibraries.md

```diff
@@ -148,12 +148,34 @@ For this purpose DPC++ front-end generates `module-id` attribute on each
 
 ### sycl-post-link changes
 
-In order to support dynamic linking of device code, `sycl-post-link` performs
-2 main tasks:
-
-- Supplies device images containing exports with an information about exported
-  symbols
-- Supplies device images with an information about imported symbols
+In typical device code splitting, all dependent device code is brought into a split
+image in order to construct a complete image.  This is impossible with dynamic linking
+since the dynamic library may rely on functions that are not defined in the library.
+Thus, when supporting dynamic linking a dependency to a canBeImportedFunction will not
+cause the dependent function to be added to the image.
+
+A canBeImportedFunction is:
+
+  1. Not an intrinsic
+  2. Name does not start with "__"
+  3. Demangled name does not start with "__"
+  4. Must be a SYCL_EXTERNAL function
+
+In order to support dynamic linking of device code, `sycl-post-link` has
+4 modifications:
```

Comment:
```suggestion
the following modifications:
```

Nit, just to avoid having to change the number if we ever decide to reword the list below somehow

Reply:
Changed